src/Interpreters/convertFieldToType.cpp

```diff
@@ -249,6 +249,25 @@ Field convertFieldToTypeImpl(const Field & src, const IDataType & type, const ID
         /// Already in needed type.
         return src;
     }
+    else if (which_type.isIPv4() && src.getType() == Field::Types::IPv4)
+    {
+        /// Already in needed type.
+        return src;
+    }
```

Comment:
Maybe add a conversion from integer fields?

Reply:
I would rather not - the accurate conversion is only possible to type UInt32 and larger - such limitation is not obvious in implicit context.